tinker_access_client/tinker_access_client/LcdApi.py

```diff
@@ -1,149 +1,229 @@
-import time
-import logging
+# -*- coding: utf-8 -*-
+# Original code found at:
+# https://gist.github.com/DenisFromHR/cc863375a6e19dce359d
+
+"""
+Compiled, mashed and generally mutilated 2014-2015 by Denis Pleic
+Made available under GNU GENERAL PUBLIC LICENSE
+
+# Modified Python I2C library for Raspberry Pi
+# as found on http://www.recantha.co.uk/blog/?p=4849
+# Joined existing 'i2c_lib.py' and 'lcddriver.py' into a single library
+# added bits and pieces from various sources
+# By DenisFromHR (Denis Pleic)
+# 2015-02-10, ver 0.1
+
+"""
 
+# Cobbled together from former TinkerAccess LcdApi.py and
+# http://www.circuitbasics.com/raspberry-pi-i2c-lcd-set-up-and-programming/
+# by kso512 (Chris Lindbergh) with lots of teaching from Erick McQueen
+# 2017-05
 
-# The wiring for the LCD is as follows:
-# 1 : GND
-# 2 : 5V
-# 3 : Contrast (0-5V)*      Tie to Gnd for max contrast
-# 4 : RS (Register Select)
-# 5 : R/W (Read Write)       - GROUND THIS PIN
-# 6 : Enable or Strobe
-# 7 : Data Bit 0             - NOT USED
-# 8 : Data Bit 1             - NOT USED
-# 9 : Data Bit 2             - NOT USED
-# 10: Data Bit 3             - NOT USED
-# 11: Data Bit 4
-# 12: Data Bit 5
-# 13: Data Bit 6
-# 14: Data Bit 7
-# 15: LCD Backlight +5V**
-# 16: LCD Backlight GND
-
-# Define GPIO to LCD mapping
-LCD_RS = 7  # 4 : RS (Register Select)  [RPi pin 7 is connected to PiWedge 'CE1' ]
-LCD_E = 8  # 6 : Enable or Strobe      [RPi pin 8 is connected to PiWedge 'CE0' ]
-LCD_D4 = 25  # 11: Data Bit 4
-LCD_D5 = 24  # 12: Data Bit 5
-LCD_D6 = 23  # 13: Data Bit 6
-LCD_D7 = 18  # 14: Data Bit 7
-
-# Define some device constants
-LCD_WIDTH = 16  # Maximum characters per line
-LCD_CHR = True
-LCD_CMD = False
-
-LCD_LINE_1 = 0x80  # LCD RAM address for the 1st line
-LCD_LINE_2 = 0xC0  # LCD RAM address for the 2nd line
-
-# Timing constants
-E_PULSE = 0.0005
-E_DELAY = 0.0005
+
+# The wiring for the LCD with backpack is as follows:
+#   LCD | Pin on Raspberry Pi
+#   ====|====================
+#   GND | 6 (GND)
+#   VCC | 4 (5V)
+#   SDA | 3 (GPIO2, SDA1 I2C)
+#   SCL | 5 (GPIO3, SCL1 I2C)
+
+
+# Start by initializing the i2c bus, then narrow down to the specific
+#   address of the LCD module:
+
+import time
+import logging
+import smbus2 as smbus
+
+# i2c bus (0 -- original Pi, 1 -- Rev 2 Pi)
+I2CBUS = 1
+
+class i2c_device:
```

Comment:
Rather than having this class, I would make these functions part of the LcdApi class. 
Just add these required constructor arguments, this way if we need to add any 'cleanup' code for when LcdApi gets disposed, we can put it in the exit function that gets called by the context manager.